lib/index.d.ts

```diff
@@ -39,3 +45,113 @@ export class StateController<State> {
   /** @internal Unsubscribe the listener callback that was passed to subscribeUpdates */
   unsubscribeUpdates(listener: (props: Partial<State>) => void): void;
 }
+
+declare namespace Component {
+    interface Helpers {
+        [constant: string]: boolean|string|number|Array|null|object;
+        [helper: string]: Function;
+    }
+
+    interface Hooks<State> {
+        /** Function called before an update is applied */
+        preUpdate?: (stateUpdate: Partial<State>) => void;
+        /** Function called after an update is applied */
+        postUpdate?: (stateUpdate: Partial<State>) => void;
+        [hookName: string]: (params: any) => void;
```

Comment:
why?

Reply:
is it for the `key of` calls below?

Reply:
you can (in theory) define arbitrary hooks

Reply:
arbitrary hooks sound dangerous, could we err on side of `less is more?`